### PR TITLE
Me: Simplify help/me main component

### PR DESCRIPTION
**Pull request by sirreal**
- Use `some()` over `!! find()`
- Leverage library function `isWpComBusinessPlan`
- Add tests
- Sort imports 

## Testing
1. Tests pass?

### CODE REVIEW

File: client/me/help/main.jsx [hunk 1]
```diff
@@ -3,34 +3,32 @@
 /**
  * External dependencies
  */
-
-import { find } from 'lodash';
-import { localize } from 'i18n-calypso';
-import React from 'react';
 import debugModule from 'debug';
+import React from 'react';
```
Comment (dmsnell) [inline client/me/help/main.jsx]: why are you reordering the React import?

File: client/me/help/main.jsx [hunk 2]
```diff
@@ -243,15 +241,16 @@ class Help extends React.PureComponent {
...
+	return isWpComBusinessPlan( productSlug );
+}
 
 export const mapStateToProps = ( state, ownProps ) => {
 	const isEmailVerified = isCurrentUserEmailVerified( state );
 	const userId = getCurrentUserId( state );
 	const purchases = getUserPurchases( state, userId );
 	const isLoading = isFetchingUserPurchases( state );
-	const isBusinessPlanUser = purchases && !! find( purchases, isWPCOMBusinessPlan );
+	const isBusinessPlanUser = some( purchases, purchaseIsWpComBusinessPlan );
```
Comment (dmsnell) [inline client/me/help/main.jsx]: after this change is there much use in keeping the function external or should we inline it here?

File: client/me/help/main.jsx (hunk 1 above)
Comment (sirreal) [reply client/me/help/main.jsx]: I reordered the imports alphabetically.

File: client/me/help/main.jsx (hunk 2 above)
Comment (sirreal) [reply client/me/help/main.jsx]: Inlining it would be good 👍

**Pull request merged**